### PR TITLE
Finalize seele(usul) deployment UI

### PR DESCRIPTION
**Pull request by mudrila**
- Adjust styling accordingly to design
- Implement a simple helper to format address in a readable format to be consistent across the app
- Adjust logic of creating transactions in order to show `multiSend` transaction on confirmation step

### CODE REVIEW

File: src/components/DAO/ExpandDAO/DeploySeele/index.tsx [hunk 1]
```diff
@@ -61,47 +56,58 @@ const DeploySeele: FunctionComponent<{
...
 			if (expandProposal) {
-				setStage("waiting")
+				// TODO: Redirect to proposal details page
+				console.log("TODO")
 			}
 		}
 	}, [proposals])
 
+	const handleProceedToConfirm = async () => {
+		if (signer) {
```
Comment (Koochr) [inline src/components/DAO/ExpandDAO/DeploySeele/index.tsx:66]: You mess up the whole data flow and make it ambiguous with this edition.
What if we go back from confirm stage?
You add the multiTx as a regular tx in the sequence, and then search for it.
I see that it's handled differently, but with this approach we can't always be sure we have this tx in right place.
There's an effect above, where we build the transaction sequence. I suggest you to create a new state for this multiTx and build it that effect each time when we add or remove a strategy (and include the case when we remove all strategies and multiTx becomes null)
Than we'll always be sure that we have it in place and it's always up-to-date

File: src/pages/Profile/index.tsx [hunk 2]
```diff
@@ -96,7 +97,7 @@ const Profile: FunctionComponent = () => {
 					</div>
 					<div className="profile__info">
 						<h2>{user.name || "Unnamed user"}</h2>
-						<p>{`${user.account.slice(0, 3)}...${user.account.slice(-4)}`}</p>
+						<p>{formatReadableAddress(user.account)}</p>
```
Comment (Koochr) [inline src/pages/Profile/index.tsx:100]: Honestly, it would be better to create a separate PR for that instead of making one that big, but not a big deal

**Review by Koochr — CHANGES_REQUESTED** [review]

File: src/components/Modals/DeployVotingStrategyModal/index.tsx [hunk 3]
```diff
@@ -1,38 +1,175 @@
...
+	const {signer} = useContext(EthersContext)
+	const [formValues, setFormValues] = useState<VotingStrategyFormValues>(EMPTY_STATE)
+
+	const {votingPeriod, delay, quorumThreshold} = formValues
+	const subTitle = VOTING_STRATEGIES.find(
+		votingStrategy => votingStrategy.strategy === strategy
+	)?.title
+	const handleTokenCreate = (token: ERC20Token) => {
+		setFormValues(prevState => ({...prevState, erc20TokenAddress: token.address}))
+		setCreateTokenModalOpened(false)
```
[the remark this answers is not in the document]
Comment (Koochr) [reply src/components/Modals/DeployVotingStrategyModal/index.tsx:40]: Got it with token address not filled out issue. This is because you forgot to pass `value` when refactoring `input`, will fix it in master

[the remark this answers is not in the document]
Comment (mudrila) [reply src/components/Modals/DeployVotingStrategyModal/index.tsx:40]: Maybe the issue was already fixed - I just re-checked token creation, it just works

File: src/components/DAO/ExpandDAO/DeploySeele/index.tsx (hunk 1 above)
Comment (mudrila) [reply src/components/DAO/ExpandDAO/DeploySeele/index.tsx:68]: Partially changed - I don't see it necessary to re-build `multiTx` each time when we add/remove strategies, but rebuilding it each time when we clicking "Continue" button - thus we doing it only when user is kind of sure about content of multiTx

File: src/components/DAO/ExpandDAO/ChooseVotingStrategies/index.tsx [hunk 4]
```diff
@@ -18,93 +22,92 @@ const ChooseVotingStrategies: FunctionComponent<{
...
-	const [removeStrategyModalOpened, setRemoveStrategyModalOpened] = useState<number | null>(null)
 
-	const handleSubmitVotingStrategy = (votingStrategy: BuiltVotingStrategy) => {
-		onStrategyAdd(votingStrategy)
-		setAddStrategyModalOpened(null)
+	const handleSubmitVotingStrategy = async (
+		strategy: VotingStrategy,
+		{erc20TokenAddress, quorumThreshold, delay, votingPeriod}: VotingStrategyFormValues
+	) => {
+		if (strategy === "linearVoting") {
```
[the remark this answers is not in the document]
Comment (mudrila) [reply src/components/DAO/ExpandDAO/ChooseVotingStrategies/index.tsx:33]: I don't see any point in having 6 modals with 50% of the same content and form fields. I would rather handle that via hooks if the validation and API calling logic will differ.
We gonna revisit this architecture anyway once the skills tree design will be rolled out, and if the form should become more complicated or differ significantly for different strategies - then for me it would make sense to split them. For now, your suggestion is premature optimization, which might not make any sense due to the upcoming revision.
WDYT to keeping it same right now and separating to different components in future, if such a need will occur?

[the remark this answers is not in the document]
Comment (Koochr) [reply src/components/DAO/ExpandDAO/ChooseVotingStrategies/index.tsx:33]: My opinion is completely opposite.
Duplication is better than bad abstraction.
We have only 1 strategy now and we don't have an idea, what API and params the other strategies would have.
Premature optimization is what you do now - you are trying to create abstraction before we have things to abstract.
We can add several strategies (and it can take quite a long) and if we see that there's a lot of duplication in our code and we can abstract it out, we will do it. This is the easier way, I think

File: src/components/DAO/ExpandDAO/DeploySeele/index.tsx (hunk 1 above)
Comment (Koochr) [reply src/components/DAO/ExpandDAO/DeploySeele/index.tsx:68]: First, don't set state with result of await call. I've even done a small research to figure out what this will lead to.
If you add a considerable delay to `buildMultiSendTx` function, the further `handleProceedToConfirm` function execution will freeze until this promise resolves. So you potentially freeze the UI until it is executed.
Second, this approach, again, messes up the data flow. `MultiTx` is a derivative of tx sequence and nothing else. You make it depend on the `stage`, which is truly redundant.
Please, get back to the approach I mentioned. You can write an async function to avoid chaining and call it in effect. First, build tx sequence, then, build multiTx, then, update the state. With this approach you can be sure the data to be always up-to-date.
And FYI, `buildMultiSendTx` is off-chain, so it's not a problem to update it with each tx sequence change.

**Review by Koochr — CHANGES_REQUESTED** [review]
First of all, please, make this PR updating just the UI. Data flow changes can be discussed separately.

File: src/components/DAO/ExpandDAO/DeploySeele/index.tsx (hunk 1 above)
Comment (mudrila) [reply src/components/DAO/ExpandDAO/DeploySeele/index.tsx:68]: > First, don't set state with result of await call.

Technically - there's no difference. We either put value to the variable or directly put that to the state. And UI might potentially freeze anyway, but the difference will be that it could happen after each voting strategy selection/removal, or only once - before proceeding to the confirmation. And the last one makes much more sense to me - we don't need "intermediate" multiTx, but the only final one. We don't need multiTx generated before actually going to confirmation without a complete set of transactions, but only with the final batch.

I would say we need 3rd opinion here.

File: src/components/DAO/ExpandDAO/ChooseVotingStrategies/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (mudrila) [reply src/components/DAO/ExpandDAO/ChooseVotingStrategies/index.tsx:33]: > We have only 1 strategy now and we don't have an idea, what API and params the other strategies would have.

Actually - we do have clue at least about params and inputs, check the following designs:

https://xd.adobe.com/view/682820cf-2729-46f3-71cc-6cb9a332fc09-1e19/screen/57ccc050-11b4-487f-a605-02953b11e0ec/specs/
https://xd.adobe.com/view/682820cf-2729-46f3-71cc-6cb9a332fc09-1e19/screen/42f99e27-37de-40c7-bd00-18cb45e4928d/specs/
https://xd.adobe.com/view/682820cf-2729-46f3-71cc-6cb9a332fc09-1e19/screen/b118604e-c505-43cd-be34-b4baed3c11da/specs/
https://xd.adobe.com/view/682820cf-2729-46f3-71cc-6cb9a332fc09-1e19/screen/a274c88c-28d9-47f4-b124-cca5ead24341/specs/
https://xd.adobe.com/view/682820cf-2729-46f3-71cc-6cb9a332fc09-1e19/screen/aa6c8304-73bf-4217-a5a1-84b2b36f694c/specs/
https://xd.adobe.com/view/682820cf-2729-46f3-71cc-6cb9a332fc09-1e19/screen/6145f9dc-5cca-4ec4-a3e4-d920dd3f19cf/specs/

API of course might differ, and final implementation - as well. But coming up from "what we have right now" - those modals and inputs there are nearly the same. Bad abstraction is worse than duplication, but this is not a bad abstraction, coz it is based on the current state of things, rather than trying to predict what it could be. 
The abstraction makes it extremely small, while duplication will increase the code base for this small modal part ~5 times.

Again, seems like we need some additional opinion on this topic

[the remark this answers is not in the document]
Comment (nginnever) [reply src/components/DAO/ExpandDAO/ChooseVotingStrategies/index.tsx:33]: There will be many strategies that we can't predict at the moment. However there are many that you can see do share the majority of their inputs. I find an abstraction for the current state of things fine for now. We can always revisit :)

File: src/components/DAO/ExpandDAO/DeploySeele/index.tsx (hunk 1 above)
Comment (nginnever) [reply src/components/DAO/ExpandDAO/DeploySeele/index.tsx:68]: multitx really is more of a final stage action, re-building it everytime a new transaction is added or removed from the multitx bundle seems a bit redundant. I'm going to merge this in the interest of time
